File: GXpert/GXpert.Web/Modules/Masters/AcademicYear/AcademicYearColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.AcademicYear")]
[BasedOnRow(typeof(AcademicYearRow), CheckNames = true)]
public class AcademicYearColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/AcademicYearColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AcademicYearRow } from "./AcademicYearRow";

export interface AcademicYearColumns {
    Id: Column<AcademicYearRow>;
    Name: Column<AcademicYearRow>;
    Description: Column<AcademicYearRow>;
    StartDate: Column<AcademicYearRow>;
    EndDate: Column<AcademicYearRow>;
    InsertDate: Column<AcademicYearRow>;
    InsertUserId: Column<AcademicYearRow>;
    UpdateDate: Column<AcademicYearRow>;
    UpdateUserId: Column<AcademicYearRow>;
    IsActive: Column<AcademicYearRow>;
}

export class AcademicYearColumns extends ColumnsBase<AcademicYearRow> {
    static readonly columnsKey = 'Masters.AcademicYear';
    static readonly Fields = fieldsProxy<AcademicYearColumns>();
}